// Copyright 2022-2024 ETSI SDG TeraFlowSDN (TFS) (https://tfs.etsi.org/)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// protocol buffers documentation: https://developers.google.com/protocol-buffers/docs/proto3
syntax = "proto3";
package vnt_manager;
import "context.proto";


service VNTManagerService {
  rpc VNTSubscript          (VNTSubscriptionRequest)    returns (VNTSubscriptionReply)  {}
  rpc ListVirtualLinkIds    (context.Empty)             returns (context.LinkIdList)    {}
  rpc ListVirtualLinks      (context.Empty)             returns (context.LinkList)      {}
  rpc GetVirtualLink        (context.LinkId)            returns (context.Link)          {}
  rpc SetVirtualLink        (context.Link)              returns (context.LinkId)        {}
  rpc RemoveVirtualLink     (context.LinkId)            returns (context.Empty)         {}
}

message VNTSubscriptionRequest {
  string host = 1;
  string port = 2;
}

message VNTSubscriptionReply {
  string subscription = 1;
}
